README.md

```diff
@@ -1,2 +1,173 @@
+- Install test dependencies:
+```bash
+pip install -r requirements-test.txt
+```
+- [A] Run the tests using pytest:
+```bash
+pytest
+```
+- [B] Run the tests using pytest with code coverage
+```bash
+pytest --cov=src --cov-report html tests/ -vv
+```
+
+## How to release
+### Update `setup.py`
+- Edit the file content with all necessary changes.
+
+### Create Distribution Files
+- Use setuptools and wheel to create distribution files. First, ensure these packages are installed:
+```bash
+pip install setuptools wheel
+```
+- Run from the project root directory:
+```bash
+python setup.py sdist bdist_wheel
+```
+- This will create the following files in a dist/ directory:
+  - A source archive (.tar.gz)
+  - A built distribution (.whl)
+  
+### Upload to PyPI
+- If not registered, create an account on PyPI.
+- Install twine to securely upload the distribution files:
+```bash
+pip install twine
+```
+- Upload Your Package:
+```bash
+twine upload dist/*
+```
+  - **You will be prompted to enter your PyPI username and password.**
+
+### Verify the Upload
+- Once uploaded, you can verify that your package is available on PyPI by visiting: https://pypi.org/project/mylibrary/
+TODO - customize the link above
+
+### Install Your Library
+- To ensure everything works as expected, try installing your library from PyPI in a clean environment:
+```bash
+pip install mylibrary
+```
+
+### Tagging the Release in Git
+```sbt
+git tag v0.1.0
+git push origin v0.1.0
+```
+
+### Contribution Guidelines
+
+We welcome contributions to the PyCompositeActionLib! Whether you're fixing bugs, improving documentation, or proposing new features, your help is appreciated.
+
+#### How to Contribute
+- **Submit Pull Requests**: Feel free to fork the repository, make changes, and submit a pull request. Please ensure your code adheres to the existing style and all tests pass.
+- **Report Issues**: If you encounter any bugs or issues, please report them via the repository's [Issues page](https://github.com/AbsaOSS/py-composite-action-lib/issues).
+- **Suggest Enhancements**: Have ideas on how to make this action better? Open an issue to suggest enhancements.
+
+Before contributing, please review our [contribution guidelines](https://github.com/AbsaOSS/py-composite-action-lib/blob/master/CONTRIBUTING.md) for more detailed information.
+
+### License Information
+
+This project is licensed under the Apache License 2.0. It is a liberal license that allows you great freedom in using, modifying, and distributing this software, while also providing an express grant of patent rights from contributors to users.
+
+For more details, see the [LICENSE](https://github.com/AbsaOSS/py-composite-action-lib/blob/master/LICENSE) file in the repository.
+
+### Contact or Support Information
+
+If you need help with using or contributing to Generate Release Notes Action, or if you have any questions or feedback, don't hesitate to reach out:
```

Comment:
Wrong copy&paste?

Reply:
You are right. Fixed in commit - 0b351127825f0dd446f0a0828295429ff5627f10.